### PR TITLE
Pmm 7449 backup location list

### PR DESCRIPTION
**Pull request by fabio-silva**
What this PR does / why we need it: Add backup page and backup locations list

Which issue(s) this PR fixes: https://jira.percona.com/browse/PMM-7449

# Implementation details
- Added page on frontend side for now;
- Backup page similar to IA;
- Adapted `Table` to allow rendering selected rows without having to do it manually with `useState` on every page where we need to see a row's details;
- The above works by adding the `useExpanded` from `react-table`. Then, a `renderExpandedRow` prop was added to the `Table`, allowing to render whatever needed when a row is selected;
- To make a cell selectable, the only thing needed is adding `{...row.getToggleRowExpandedProps()}` to the component returned from a a column's `Cell` method (the method's `row` argument includes an `isExpanded` flag);
- Create a `SecretToggler` component, which allows to switch between a text/password read-only input field, using the lock icon on the right.

# Screenshots
Screenshot 2021-02-15 at 10 48 50

### CODE REVIEW

File: public/app/features/backup/components/DescriptionBlock/DescriptionBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { getStyles } from './DescriptionBlock.styles';
+
+export const DescriptionBlock: FC<DescriptionBlockProps> = ({ description }) => {
+  const styles = useStyles(getStyles);
+
+  return (
+    <div className={styles.descriptionWrapper}>
+      <span>{Messages.description}</span>
+      <pre>{description}</pre>
+    </div>
```
Comment (nicolalamacchia) [inline public/app/features/backup/components/DescriptionBlock/DescriptionBlock.tsx:14]: You should consider adding `data-qa` attributes here

Comment (fabio-silva) [reply public/app/features/backup/components/DescriptionBlock/DescriptionBlock.tsx:14]: done 👍

File: grafana-mixin/dashboards/grafana-overview.json [hunk 2]
```diff
@@ -55,9 +55,7 @@
         "justifyMode": "auto",
```
Comment (tiagomotasantos) [inline grafana-mixin/dashboards/grafana-overview.json:55]: Why this changes are in this PR? Was it on purpose?

File: public/app/features/backup/components/DBIcon/DBIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React, { FC } from 'react';
+import { toPascalCase } from '@grafana/data';
+import { DBIconProps } from './DBIcon.types';
+import * as Icons from './assets';
+
+export const DBIcon: FC<DBIconProps> = ({ type, size, ...rest }) => {
+  //@ts-ignore
```
Comment (tiagomotasantos) [inline public/app/features/backup/components/DBIcon/DBIcon.tsx:7]: Is this really required? Can't we just cast the return of `Icons` or the return of `toPascalCase` depending on the error we get.

File: public/app/features/backup/components/SecretToggler/SecretToggler.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { Icon, useStyles } from '@grafana/ui';
+import React, { FC, useState, useMemo } from 'react';
+import { SecretTogglerProps } from './SecretToggler.types';
+import { getStyles } from './SecretToggler.styles';
+
+export const SecretToggler: FC<SecretTogglerProps> = ({ secret, readOnly }) => {
```
Comment (tiagomotasantos) [inline public/app/features/backup/components/SecretToggler/SecretToggler.tsx:6]: Good one!! :)

File: public/app/features/backup/components/StorageLocations/StorageLocationsActions/StorageLocationsActions.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import { StorageLocatationsActionProps } from './StorageLocationsActions.types';
+import { getStyles } from './StorageLocationsActions.styles';
+
+export const StorageLocationsActions: FC<StorageLocatationsActionProps> = ({ location }) => {
+  const styles = useStyles(getStyles);
+
+  return (
+    <div className={styles.actionsWrapper}>
+      <DBIcon type="edit" data-qa="edit-storage-location-button" role="button" />
+      {/* <DBIcon type="see" data-qa="see-storage-location-button" role="button" /> */}
```
Comment (tiagomotasantos) [inline public/app/features/backup/components/StorageLocations/StorageLocationsActions/StorageLocationsActions.tsx:13]: Remove this if it's not needed now.

File: public/app/features/integrated-alerting/components/Alerts/Alerts.tsx [hunk 6]
```diff
@@ -1,4 +1,128 @@
...
+import { Messages } from '../../IntegratedAlerting.messages';
+import { getStyles } from './Alerts.styles';
+import { Alert, AlertStatus } from './Alerts.types';
+import { formatAlerts, getSeverityColors } from './Alerts.utils';
+import { AlertsService } from './Alerts.service';
+import { AlertRuleSeverity } from '../AlertRules/AlertRules.types';
+import { AlertsActions } from './AlertsActions';
 
-export const Alerts: FC = () => <AlertsTable />;
+const { noData, columns } = Messages.alerts.table;
```
Comment (tiagomotasantos) [inline public/app/features/integrated-alerting/components/Alerts/Alerts.tsx:15]: Why this big changes to `Alerts` in this PR about backups? 🤔

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]
I would just center the "Description" label in the table. The rest looks good to me good job.

File: grafana-mixin/dashboards/grafana-overview.json (hunk 2 above)
Comment (fabio-silva) [reply grafana-mixin/dashboards/grafana-overview.json]: The branch was based on the table refactor, which got merged meanwhile, so GH was comparing against master. Now it's updated with master again.

File: public/app/features/integrated-alerting/components/Alerts/Alerts.tsx (hunk 6 above)
Comment (fabio-silva) [reply public/app/features/integrated-alerting/components/Alerts/Alerts.tsx]: Same reason as above ;)

File: public/app/features/backup/components/DBIcon/DBIcon.tsx (hunk 3 above)
Comment (fabio-silva) [reply public/app/features/backup/components/DBIcon/DBIcon.tsx]: seems like TS really doesn't like wildcard imports (unless casting `Icon` as `any`), so I went the other way around with a map, to play safe.

File: public/app/features/backup/components/StorageLocations/StorageLocationsActions/StorageLocationsActions.tsx (hunk 5 above)
Comment (fabio-silva) [reply public/app/features/backup/components/StorageLocations/StorageLocationsActions/StorageLocationsActions.tsx]: 👍

**Review by tiagomotasantos — APPROVED** [review]

**Comment by fabio-silva** [comment]
Added S3's bucket name to the fields, as per https://jira.percona.com/browse/PMM-7573

**Review by tiagomotasantos — APPROVED** [review]
